src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -154,27 +154,39 @@ Assembly[] LoadConfigurationAssemblies()
                 }
             }
 
-            foreach (var library in _dependencyContext.RuntimeLibraries)
+            foreach (var assemblyName in GetSerilogConfigurationAssemblies())
             {
-                if (library.Name != null && library.Name.ToLowerInvariant().Contains("serilog"))
-                {
-                    var assumedName = new AssemblyName(library.Name);
-                    var assumed = Assembly.Load(assumedName);
-                    if (assumed != null && !assemblies.ContainsKey(assumed.FullName))
-                        assemblies.Add(assumed.FullName, assumed);
-
-                    foreach (var assemblyRef in library.Assemblies)
-                    {
-                        var assembly = Assembly.Load(assemblyRef.Name);
-                        if (assembly != null && !assemblies.ContainsKey(assembly.FullName))
-                            assemblies.Add(assembly.FullName, assembly);
-                    }
-                }
+                var assumedName = new AssemblyName(assemblyName);
+                var assumed = Assembly.Load(assumedName);
+                if (assumed != null && !assemblies.ContainsKey(assumed.FullName))
+                    assemblies.Add(assumed.FullName, assumed);
             }
 
             return assemblies.Values.ToArray();
         }
 
+        string[] GetSerilogConfigurationAssemblies()
+        {
+            var query = Enumerable.Empty<string>();
+            var filter = new Func<string, bool>(name => name != null && name.ToLowerInvariant().Contains("serilog"));
+
+            if (_dependencyContext != null)
+            {
+                query = from lib in _dependencyContext.RuntimeLibraries where filter(lib.Name) select lib.Name;
+            }
+            else
+            {
+#if APPDOMAIN
+                query = from outputAssemblyPath in System.IO.Directory.GetFiles(AppDomain.CurrentDomain.BaseDirectory, "*.dll", System.IO.SearchOption.AllDirectories)
+                        let assemblyFileName = System.IO.Path.GetFileNameWithoutExtension(outputAssemblyPath)
+                        where filter(assemblyFileName)
+                        select Assembly.ReflectionOnlyLoadFrom(outputAssemblyPath).GetName().FullName;
+#endif
```

Comment:
I think dependency context (and requirement to have preserveCompiltationContext=true) should be optional for several reasons:
1) it is already optional on public API (LoggerSettingsConfiguration.Configuration(..))
2) if some custom build tool doesn't generate dependency context, this library still has "Using" section, where we can put configuration assemblies without restriction to use this library (the source of this issue).
3) even if   dependencyContext is not null it doesn't help with non-conventional config assemblies that don't have "serilog" name in it. Again, we can put them in "Using" section.